datafusion/physical-expr-adapter/src/schema_rewriter.rs

```diff
@@ -220,13 +223,111 @@ impl<'a> DefaultPhysicalExprAdapterRewriter<'a> {
         &self,
         expr: Arc<dyn PhysicalExpr>,
     ) -> Result<Transformed<Arc<dyn PhysicalExpr>>> {
+        if let Some(transformed) = self.try_rewrite_struct_field_access(&expr)? {
+            return Ok(Transformed::yes(transformed));
+        }
+
         if let Some(column) = expr.as_any().downcast_ref::<Column>() {
             return self.rewrite_column(Arc::clone(&expr), column);
         }
 
         Ok(Transformed::no(expr))
     }
 
+    fn try_rewrite_struct_field_access(
+        &self,
+        expr: &Arc<dyn PhysicalExpr>,
+    ) -> Result<Option<Arc<dyn PhysicalExpr>>> {
+        let get_field_expr = match expr.as_any().downcast_ref::<ScalarFunctionExpr>() {
+            Some(expr) => expr,
+            None => return Ok(None),
+        };
+
+        if get_field_expr.name() != "get_field" {
```

Comment:
Why do we need to check the function name. Isn't the check for downcast_ref().is_none sufficient?

Also, it seems to me that this rewrite is specific to `GetFieldFunc` so I think it might make sense, long term, to put the function alongside the GetFieldFunc  somehow 🤔 

Maybe we could add a trait to PhysicalExpr like `rewrite_for_schema` or something 🤔

Reply:
I've seen / used this pattern a lot. I added 50b1e91ef which should simplify it not just for this PR but any other use of matching a specific function from a PhysicalExpr as well.